src/ml_gym/frontend/dashboard/src/routes/flow_board.tsx

```diff
@@ -0,0 +1,122 @@
+import { AgGridReact } from "ag-grid-react";
+import React, { Component } from "react";
+import type { RootState } from '../app/store';
+import 'ag-grid-community/styles/ag-grid.css'; // Core grid CSS, always needed
+import 'ag-grid-community/styles/ag-theme-alpine.css'; // Optional theme CSS
+import { jobStatusRowsSelector } from '../features/jobsStatus/jobsStatusSlice'
+import { connect } from "react-redux";
+
+import { JobStatusRowType, ModelStatusRowType } from '../app/datatypes'
+
+type FlowBoardPropsType = {
+    jobStatusRows: any
+};
+
+
+
+const statusRowsSelector = (state: RootState) => {
+    const jobStatusRows = jobStatusRowsSelector(state)
+
+    const fullRows = jobStatusRows.map((jobStatusRow: JobStatusRowType) => {
+        const latest_message_index = state.modelsStatus.experiment_id_to_latest_message_index[jobStatusRow.experiment_id]
+        if (latest_message_index) {
+            const modelStatusMessage = state.modelsStatus.messages[latest_message_index]
+            const modelStatusRow = {
+                experiment_id: modelStatusMessage.data.payload.experiment_id,
+                model_status: modelStatusMessage.data.payload.status,
+                num_epochs: modelStatusMessage.data.payload.num_epochs,
+                current_epoch: modelStatusMessage.data.payload.current_epoch,
+                splits: modelStatusMessage.data.payload.splits,
+                current_split: modelStatusMessage.data.payload.current_split,
+                epoch_progress: modelStatusMessage.data.payload.current_epoch / modelStatusMessage.data.payload.num_epochs,
```

Comment:
Codacy has a fix for the issue: [Replace `················epoch_progress:·modelStatusMessage.data.payload.current_epoch·/` with `········epoch_progress:⏎··········modelStatusMessage.data.payload.current_epoch·/⏎·········`](https://app.codacy.com/gh/mlgym/mlgym/pullRequest?prid=9966992)
```suggestion
        epoch_progress:
```